### ISSUE TITLE
TEMP: Final comment period

### ISSUE THREAD

**Issue by philpax** [goatcorp/DIPs#2]
I am adapting the Rust RFC process to DIPs, and am choosing to leave this language out:

> - At some point, a member of the subteam will propose a "motion for final
>   comment period" (FCP), along with a _disposition_ for the DIP (merge, close,
>   or postpone).
>   - This step is taken when enough of the tradeoffs have been discussed that
>     the subteam is in a position to make a decision. That does not require
>     consensus amongst all participants in the DIP thread (which is usually
>     impossible). However, the argument supporting the disposition on the DIP
>     needs to have already been clearly articulated, and there should not be a
>     strong consensus _against_ that position outside of the subteam. Subteam
>     members use their best judgment in taking this step, and the FCP itself
>     ensures there is ample time and notification for stakeholders to push back
>     if it is made prematurely.
>   - For DIPs with lengthy discussion, the motion to FCP is usually preceded by
>     a _summary comment_ trying to lay out the current state of the discussion
>     and major tradeoffs/points of disagreement.
>   - Before actually entering FCP, _all_ members of the subteam must sign off;
>     this is often the point at which many subteam members first review the DIP
>     in full depth.
> - The FCP lasts ten calendar days, so that it is open for at least 5 business
>   days. It is also advertised widely. This way all stakeholders have a chance
>   to lodge any final objections before a decision is reached.
> - In most cases, the FCP period is quiet, and the DIP is either merged or
>   closed. However, sometimes substantial new arguments or ideas are raised,
>   the FCP is canceled, and the DIP goes back into development mode.

I don't think we have enough formal organisation / machinery to do this, and I'm not sure if it's desirable yet.

**Comment by philpax** [comment]
We've had a little more experience with DIPs now, and it seems like having a FCP would be worthwhile. That being said, I also don't really want to institute it without some of the automated bookkeeping (like #30) - otherwise, we run the risk of DIPs going through without stakeholders being appropriately notified.

Probably need to study the Rust process some more here.

**Comment by philpax** [comment]
so for our two big boy DIPs so far, #17 and #18, I've been instituting an informal FCP of a week and just notifying the relevant people. would be nice to automate this (maybe with bors?) but honestly this is fine for now

**Comment by philpax** [comment]
Yep, things stagnate without a bot to push things along. @NotNite how hard would it be to get bleatbot to do this?

**Comment by NotNite** [comment]
It shouldn't be too hard. I might take a crack at implementing it.
